core/markets/retrieval/provider/retrieval_provider_types.hpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_MARKETS_RETRIEVAL_PROVIDER_TYPES_HPP
+#define CPP_FILECOIN_CORE_MARKETS_RETRIEVAL_PROVIDER_TYPES_HPP
+
+#include <string>
+
+#include "markets/retrieval/types.hpp"
+#include "primitives/types.hpp"
+
+namespace fc::markets::retrieval::provider {
+  using primitives::TokenAmount;
+
+  /**
+   * @struct Provider config
+   */
+  struct ProviderConfig {
+    TokenAmount price_per_byte;
+    uint64_t payment_interval;
+    uint64_t interval_increase;
+  };
+
+  /**
+   * @struct Provider deal state
+   */
+  struct ProviderDealState {
+    /* Current deal status */
+    DealStatus deal_status;
+
+    /* Sender PeerID */
+    uint64_t sender_peer_id;
```

Comment:
It's not `uint64_t`, use `libp2p::peer::PeerId`.
Include "common/libp2p/peer/cbor_peer_id.hpp" for cbor.